operator/v1/types_ingress.go

```diff
@@ -1346,6 +1346,78 @@ type IngressControllerHTTPHeaders struct {
+
+	// actions specifies options for performing action on headers.
+	// For Setting Headers you can provide a value for the header whereas for deleting one needs to provide header name.
+	// +optional
+	Actions IngressControllerHTTPHeaderActions `json:"actions,omitempty"`
+}
+
+// IngressControllerHTTPHeaderActions defines whether we want to set/delete Response or Request Headers.
+type IngressControllerHTTPHeaderActions struct {
+	// Response is a list of http response headers which either need to set or deleted.
+	Response []IngressControllerHTTPHeader `json:"response,omitempty"`
+	// Request is a list of http request headers which either need to set or deleted.
+	Request []IngressControllerHTTPHeader `json:"request,omitempty"`
+}
+
+// IngressControllerSetOrDeleteHTTPHeader specifies options for set/delete a header.
+type IngressControllerHTTPHeader struct {
+	// name specifies a header name to be set/delete.  Its value must be a valid HTTP header.
+	// name as defined in RFC 2616 section 4.2.
+	// +required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=256
+	// +kubebuilder:validation:Pattern="^[-!#$%&'*+.0-9A-Z^_`a-z|~]+$"
+	Name string `json:"name"`
+
+	// value specifies fields to mention the header details and the action you want to perform on them.
+	Value IngressControllerHTTPHeaderActionUnion `json:"value"`
+}
+
+// IngressControllerHTTPHeaderActionUnion specifies fields to mention the header details and the action you want to perform on them.
+// +union
+type IngressControllerHTTPHeaderActionUnion struct {
+	// action defines the type of the action to be applied on the header.
+	// Set allows you to set http request and response headers.
+	// Delete allows you to delete http request and response headers
+	// +unionDiscriminator
+	// +kubebuilder:validation:Enum:=Set;Delete
+	// +required
+	Action IngressControllerHTTPHeaderActionType `json:"action,omitempty"`
+
+	// set defines HTTP headers that should be set.
+	// If this field is empty, no headers are added.
+	//
+	// Note that this option only applies to cleartext HTTP connections
+	// and to secure HTTP connections for which the ingress controller
+	// terminates encryption (that is, edge-terminated or reencrypt
+	// connections).  Headers cannot be set for TLS passthrough
+	// connections.
+	//
+	// +optional
+	// +unionMember
+	Set *IngressControllerSetHTTPHeader `json:"set,omitempty"`
+}
+
+// IngressControllerHTTPHeaderActionType defines operation you can perform on headers.
+type IngressControllerHTTPHeaderActionType string
+
+const (
+	// Set is used specify set action per header
+	Set IngressControllerHTTPHeaderActionType = "Set"
+	// Delete is used specify delete action per header
+	Delete IngressControllerHTTPHeaderActionType = "Delete"
+)
+
+// IngressControllerSetHTTPHeader defines the value which needs to be set on an HTTP header
+type IngressControllerSetHTTPHeader struct {
+	// value specifies a header value.
```

Comment:
Document that the value is interpreted as an HAProxy format string as defined in http://cbonte.github.io/haproxy-dconv/2.6/configuration.html#8.2.6 and may use HAProxy's `%[]` syntax and otherwise must be a valid HTTP header value as defined in https://datatracker.ietf.org/doc/html/rfc7230#section-3.2.

Reply:
@Miciah Are you happy that this has been covered?

Reply:
Yes, I'm sufficiently happy.  I'd prefer we cite the HAProxy 2.6 documentation <http://cbonte.github.io/haproxy-dconv/2.6/configuration.html#8.2.6> rather than the HAProxy 2.4 documentation <https://cbonte.github.io/haproxy-dconv/2.4/configuration.html#8.2.4>, but the version isn't really important; the relevant content hasn't changed.

Reply:
@Miciah, @candita was concerned about the version

Reply:
But HAProxy 2.6 is the correct version for OpenShift 4.14, and HAProxy 2.4 is the wrong version for any OpenShift release—we have never shipped HAProxy 2.4.

Reply:
ok I did add the 2.6 version earlier